### ISSUE TITLE
inconsistent of the algorithm in the paper and the implementation.

### ISSUE THREAD

**Issue by mayorx** [YingZhangDUT/Deep-Mutual-Learning#2]
I'm a little confused.
In your paper, the 3rd step of Algorithm 1 is "Update the predictions p1 && p2 ..."
but the implementation [here](https://github.com/YingZhangDUT/Deep-Mutual-Learning/blob/master/train_models.py#L94) doesn't update the predicts.

**Comment by YingZhangDUT** [comment]
Hi, each iteration we compute the p1 and p2 to calculate KL loss, which means "update the prediction of p1/p2".

**Comment by mayorx** [comment]
ok, thanks for your reply.

**Comment by Guo-Xiaoqing** [comment]
Thanks for your reply! For this question, I understand now. Thanks! 

Did you use mobileNet to classify the Cifar100? 

I don't understand why he try to computer grads and apply grades to every parameters, which seems to be a little complex. Can I use tf.train.AdamOptimizer(FLAGS.learning_rate,
                                                                  beta1=FLAGS.adam_beta1,
                                                                  beta2=FLAGS.adam_beta2,
                                                                  epsilon=FLAGS.opt_epsilon).minimize(loss, var_list) directly?
